sentry-ruby/lib/sentry/faraday.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+module Sentry
+  module Faraday
+    OP_NAME = "http.client"
+    SPAN_ORIGIN = "auto.http.faraday"
+    BREADCRUMB_CATEGORY = "http"
+
+    module Connection
+      # Since there's no way to preconfigure Faraday connections and add our instrumentation
+      # by default, we need to extend the connection constructor and do it there
+      #
+      # @see https://lostisland.github.io/faraday/#/customization/index?id=configuration
+      def new(url = nil, options = nil)
+        super.tap do |connection|
+          # Ensure that we attach instrumentation only if the adapter is not net/http
+          # because if is is, then the net/http instrumentation will take care of it
+          if connection.builder.adapter.name != "Faraday::Adapter::NetHttp"
+            connection.request :instrumentation, name: OP_NAME, instrumenter: Instrumenter.new
+
+            # Make sure that it's going to be the first middleware so that it can capture
+            # the entire request processing involving other middlewares
+            handlers = connection.builder.handlers
+            instrumentation = handlers.pop
+
+            handlers.unshift(instrumentation)
```

Comment:
`handlers` actually exposes an `insert` method (which is more low level than `use` sadly) so maybe we can change it later and avoid the array weirdness but it's fine for now this way 
https://github.com/lostisland/faraday/blob/4860f75372bdfac5612c76d76bd713e72b3f45b3/lib/faraday/rack_builder.rb#L118-L123